packages/core/src/2d/sprite/SpriteMask.ts

```diff
@@ -185,10 +168,7 @@ export class SpriteMask extends Renderer implements ICustomClone {
   _onDestroy(): void {
     this._sprite = null;
     this._renderData = null;
-    if (this._spriteChangeFlag) {
-      this._spriteChangeFlag.destroy();
-      this._spriteChangeFlag = null;
-    }
+    this._sprite?._updateFlagManager.removeListener(this._onSpriteChange);
```

Comment:
Should not be after `this._sprite = null; ` here.

Reply:
done